### PR TITLE
ApiManagement - revert required property in openidconnectprovider

### PR DESCRIPTION
**Pull request by vfedonkin**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by solankisamir — CHANGES_REQUESTED** [review]
:clock1:

**Review by solankisamir — APPROVED** [review]
:shipit:

**Comment by vfedonkin** [comment]
@erich-wang , could you please merge this pull request?

**Comment by erich-wang** [comment]
@solankisamir, we need to wait someone from ARM API review team to approve this pr, I've sent email to ping them.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json [hunk 1]
```diff
@@ -2878,54 +2878,6 @@
...
-          "x-ms-secret": true,
-          "type": "string",
-          "description": "Client Secret of developer console which is the client application."
-        }
-      },
-      "required": [
-        "displayName",
-        "metadataEndpoint",
-        "clientId",
-        "clientSecret"
```
Comment (ryansbenson) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json]: The only actual change here is 'clientSecret' is no longer required. Was this intended? Is this how the service works or is this a change to the API being introduced?

Comment (vfedonkin) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json]: @ryansbenson , it's intended and how the service works.   We just marked the clientSecret as required by mistake in one of previous PRs, so now we revert it back.

**Comment by vfedonkin** [comment]
Ping! Can anybody merge this PR, please?

**Comment by solankisamir** [comment]
@ryansbenson can you take a look at this again? This is blocking multiple downstream projects.

**Comment by solankisamir** [comment]
@sanjaiganesh a gentle ping on this PR

**Pull request merged**